app/src/main/java/com/infomaniak/mail/utils/MessageBodyUtils.kt

```diff
@@ -51,27 +52,32 @@ object MessageBodyUtils {
         "[name=\"quote\"]", // GMX
     )
 
-    fun splitContentAndQuote(body: Body): SplitBody {
+    fun splitContentAndQuote(body: Body): SplitBody = runBlocking(Dispatchers.IO) {
 
-        if (body.type == Utils.TEXT_PLAIN) return SplitBody(body.value)
+        if (body.type == Utils.TEXT_PLAIN) return@runBlocking SplitBody(body.value)
 
-        // Heuristic to give up on mail too large for "perfect" preprocessing.
-        if (body.value.toByteArray().size > QUOTE_DETECTION_SIZE_LIMIT) return SplitBody(body.value)
+        return@runBlocking runCatching {
 
-        // The original parsed html document in full
-        val originalHtmlDocument = Jsoup.parse(body.value)
-        // Initiated to the original document and it'll be processed by Jsoup to remove quotes.
-        val htmlDocumentWithoutQuote = originalHtmlDocument.clone()
+            withTimeout(QUOTE_DETECTION_TIMEOUT) {
 
-        // Find the last parent blockquote and delete it in htmlDocumentWithoutQuote
-        val blockquoteElement = findAndRemoveLastParentBlockquote(htmlDocumentWithoutQuote)
-        // Find the first known parent quote in the html and delete all known quotes descriptor
-        val currentQuoteDescriptor = findFirstKnownParentQuoteDescriptor(htmlDocumentWithoutQuote).ifEmpty {
-            if (blockquoteElement == null) "" else BLOCKQUOTE
-        }
+                // The original parsed HTML document in full
+                val originalHtmlDocument = Jsoup.parse(body.value)
+                // Initiated to the original document and it'll be processed by Jsoup to remove quotes
+                val htmlDocumentWithoutQuote = originalHtmlDocument.clone()
+
+                // Find the last parent blockquote and delete it in `htmlDocumentWithoutQuote`
+                val blockquoteElement = findAndRemoveLastParentBlockquote(htmlDocumentWithoutQuote)
+                // Find the first known parent quote in the HTML and delete all known quotes descriptor
+                val currentQuoteDescriptor = findFirstKnownParentQuoteDescriptor(htmlDocumentWithoutQuote, scope = this).ifEmpty {
+                    if (blockquoteElement == null) "" else BLOCKQUOTE
+                }
 
-        val (content, quote) = splitContentAndQuote(originalHtmlDocument, currentQuoteDescriptor, blockquoteElement)
-        return if (quote.isNullOrBlank()) SplitBody(body.value) else SplitBody(content, body.value)
+                val (content, quote) = splitContentAndQuote(originalHtmlDocument, currentQuoteDescriptor, blockquoteElement)
+                if (quote.isNullOrBlank()) SplitBody(body.value) else SplitBody(content, body.value)
+            }
+        }.getOrElse {
+            SplitBody(body.value)
+        }
```

Comment:
If we get another exception than TimeoutCancellationException we will never know it

Reply:
Also, we should probably send a Sentry to know how many emails triggered the Timeout. For all we know, it might not happen anymore since the last update that removed the detection of outlook quotes